contracts/exchange/contracts/src/MixinMatchOrders.sol

```diff
@@ -263,6 +259,446 @@ contract MixinMatchOrders is
+        // the profit will be out in the right maker asset.
+        //
+        // There are three cases to consider:
+        // Case 1.
+        //   If the left maker can buy more than or equal to the right maker can sell, then only the right order is fully filled.
+        // Case 2.
+        //   If the right maker can buy more than or equal to the left maker can sell, then only the right order is fully filled.
+        // Case 3.
+        //   If the right maker can sell the max of what the left maker can buy and the left maker can sell the max of
+        //   what the right maker can buy, then both orders are fully filled.
+        if (leftTakerAssetAmountRemaining >= rightMakerAssetAmountRemaining) {
+            // Case 1: Right order is fully filled with the profit paid in the left makerAsset
+            _calculateCompleteRightFill(
+                matchedFillResults,
+                leftOrder,
+                rightMakerAssetAmountRemaining,
+                rightTakerAssetAmountRemaining
+            );
+            // Indicate that the profit should be set to the spread denominated in the left maker asset.
+            doesLeftMakerAssetProfitExist = true;
+        } else if (rightTakerAssetAmountRemaining >= leftMakerAssetAmountRemaining) {
+            // Case 2: Left order is fully filled with the profit paid in the right makerAsset.
+            matchedFillResults.left.makerAssetFilledAmount = leftMakerAssetAmountRemaining;
+            matchedFillResults.left.takerAssetFilledAmount = leftTakerAssetAmountRemaining;
+            // Round down to ensure the right maker's exchange rate does not exceed the price specified by the order.
+            // We favor the right maker when the exchange rate must be rounded and the profit is being paid in the
+            // right maker asset.
+            matchedFillResults.right.makerAssetFilledAmount = _safeGetPartialAmountFloor(
+                rightOrder.makerAssetAmount,
+                rightOrder.takerAssetAmount,
+                leftMakerAssetAmountRemaining
+            );
+            matchedFillResults.right.takerAssetFilledAmount = leftMakerAssetAmountRemaining;
+            // Indicate that the profit should be set to the spread denominated in the right maker asset.
+            doesRightMakerAssetProfitExist = true;
+        } else {
+            // Case 3: The right and left orders are fully filled
+            matchedFillResults.right.makerAssetFilledAmount = rightMakerAssetAmountRemaining;
+            matchedFillResults.right.takerAssetFilledAmount = rightTakerAssetAmountRemaining;
+            matchedFillResults.left.makerAssetFilledAmount = leftMakerAssetAmountRemaining;
+            matchedFillResults.left.takerAssetFilledAmount = leftTakerAssetAmountRemaining;
+            // Indicate that the profit should be set to the spread denominated in the left and the right maker assets.
+            doesLeftMakerAssetProfitExist = true;
```

Comment:
If we do make the other statements `>` then this would become:
```
doesLeftMakerAssetProfitExist = rightTakerAssetAmountRemaining < leftMakerAssetAmountRemaining;
doesRightMakerAssetProfitExist = leftTakerAssetAmountRemaining < rightMakerAssetAmountRemaining;
```